packages/medusa/src/interfaces/file-service.ts

```diff
@@ -0,0 +1,79 @@
+import stream from "stream"
+
+export type FileServiceUploadResult = {
+  url: string
+}
+
+export type FileServiceGetUploadStreamResult = {
+  writeStream: stream.PassThrough
+  promise: Promise<any>
+  url: string
+}
+
+export interface IFileService {
+  /**
+   * upload file to fileservice
+   * @param file Multer file from express multipart/form-data
+   * */
+  upload(file: Express.Multer.File): Promise<FileServiceUploadResult>
+
+  /**
+   * remove file from fileservice
+   * @param fileData Remove file described by record
+   * */
+  delete(fileData: Record<string, any>): void
+
+  /**
+   * upload file to fileservice from stream
+   * @param fileData file metadata relevant for fileservice to create and upload the file
+   * @param fileStream readable stream of the file to upload
+   * */
+  getUploadStream(
+    fileData: Record<string, any>
+  ): Promise<FileServiceGetUploadStreamResult>
+
+  /**
+   * download file from fileservice as stream
+   * @param file file metadata relevant for fileservice to download the file
+   * @returns readable stream of the file to download
+   * */
+  getDownloadStream(file: Record<string, any>): Promise<NodeJS.ReadableStream>
```

Comment:
question: `downloadAsStream` instead of the actual name?

Reply:
definitley non-blocking: I actually like the `getDownloadStream` a bit more and think it's more intuitive than `downloadAsStream`. I prefer the first variation because `downloadAsStream` sounds like it performs some side effect without necessarily returning a stream 😅 . What do you guys think @adrien2p @pKorsholm 😄 ?

Reply:
Yeah true, I'm fine with both, but i see the case that we are not actually downloading the contents until we consume the stream later.

Reply:
It has been updated 😄